### PR TITLE
Added jsonp response.

### PR DESCRIPTION
**Pull request by kat3su**
Allow send_response() to output as jsonp format

### CODE REVIEW

**Comment by lode** [comment]
Nice, thanks @lthh89vt!

There's a few changes I would like to make to stick to a coding convention. If you want to make them, lovely. Otherwise, I'm fine doing them when I merge this in. Let me know.
- Use the `$content_type` to switch to jsonp content type (except for debug situations).
- Use an early return for jsonp instead of a ternary if.
- Use concatenation instead for building the jsonp response.

**Comment by kat3su** [comment]
Thanks, @lode 

Good that you like the idea. Will make the changes early next week.

Chs,

File: src/response.php [hunk 1]
```diff
@@ -141,9 +148,18 @@ public function send_response($content_type=null, $encode_options=null, $respons
...
+	} 
 	
-	echo $response;
+	if ($content_type == self::CONTENT_TYPE_JSONP) {
+		if (empty($jsonp_callback)) {
+			$jsonp_callback = self::JSONP_CALLBACK_DEFAULT;
+		}
+		// jsonp response
+		echo $jsonp_callback.'('.$response.')';
+	} else {
```
Comment (lode) [inline src/response.php]: Please keep the main flow unindented, by returning after echo'ing the jsonp response.

File: src/response.php [hunk 2]
```diff
@@ -141,9 +148,18 @@ public function send_response($content_type=null, $encode_options=null, $respons
 	
 	if ($this->http_status == self::STATUS_NO_CONTENT) {
 		return;
-	}
+	} 
 	
-	echo $response;
+	if ($content_type == self::CONTENT_TYPE_JSONP) {
+		if (empty($jsonp_callback)) {
+			$jsonp_callback = self::JSONP_CALLBACK_DEFAULT;
```
Comment (lode) [inline src/response.php:156]: So this generates a response of `JSON_CALLBACK({"foo":"bar"})`? I'm not so much into jsonp, but isn't there a common callback name? I guess capitalized and underscored is not really the default callback name :)

**Comment by kat3su** [comment]
I'm using angular2 so the default callback is JSONP_CALLBACK
But definitely can put default as "callback"

**Comment by lode** [comment]
Ok, that's fine with me. We can always change it if another more generic standard comes by, but using the angular one seems fine to me.

Thanks for adding this!

**Comment by lode** [comment]
I've just released [v1.3.0](https://github.com/lode/jsonapi/releases/tag/v1.3.0) to add this.

**Pull request merged**